pkg/controller/installation/core_controller.go

```diff
@@ -746,30 +754,46 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		reqLogger.Error(err, "An error occurred when querying the Installation resource")
 		return reconcile.Result{}, err
 	}
-	status := instance.Status
+
 	terminating := (instance.DeletionTimestamp != nil)
 	preDefaultPatchFrom := client.MergeFrom(instance.DeepCopy())
 
 	// Mark CR found so we can report converter problems via tigerastatus
 	r.status.OnCRFound()
 
+	// Changes for updating installation status conditions
+	if request.Name == InstallationName && request.Namespace == "" {
+		ts := &operator.TigeraStatus{}
+		err := r.client.Get(ctx, types.NamespacedName{Name: InstallationName}, ts)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		instance.Status.Conditions = status.UpdateStatusCondition(instance.Status.Conditions, ts.Status.Conditions, instance.GetGeneration())
```

Comment:
It seems wrong to use the just read generation of the instance. There is nothing ensuring that the generation we just read matches with the generation when the Status was updated. As it is written the generation could very possibly be newer than generation that has been applied and TigeraStatus represents.
Has this been hashed out and the generation not being connected to the TigeraStatus update is ok?

I think it makes sense to pass the generation from the controller to `r.status` anytime we're doing a `r.status.SetDegraded` or when we call `r.status.ReadyToMonitor()`, then the update to the TigeraStatus should include a generation and then when the controller gets the TigeraStatus we use the generation embedded there.

Reply:
Done.. invoking SetMetaData in defer